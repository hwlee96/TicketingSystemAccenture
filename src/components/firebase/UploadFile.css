#uploader {
  position: fixed;
  bottom: 0;
}

.buttonStyleUpload {
  background: #187df4;
  border-radius: 4px;
  color: #ffffff;
  font-family: "Lato";
  font-size: 20px;
  font-weight: 400;

  font-size: 80%;
  padding: 10px;
  display: block;
  cursor: pointer;
  border-style: solid;
  border: none;
  margin: 2%;
  margin-left: 87%;
}

.buttonStyleCross {
  float: right;
  cursor: pointer;
}

.successfulFileAttach {
  margin-left: 35%;
  font-size: 90%;
  color: #000000;
}

.imageAttachment {
  margin-left: 33%;
}

/*
.buttonStyleBack {
  background: #d00000;
  border-radius: 4px;
  color: #ffffff;
  font-family: "Lato";
  font-weight: 400;
  text-transform: uppercase;
  font-size: 80%;
  padding: 10px;
  display: inline-block;
  cursor: pointer;
  border-style: solid;
  border: none;
}

.btn-file-input {
  display: inline-block;
  width: 200px;
  height: 40px;
  overflow: hidden;
  position: relative;
}
.btn-file-input > input[type="file"] {
  font-size: 200px;
  position: absolute;
  top: 0;
  right: 0;
  opacity: 0;
} */

.file-input-wrapper {
  width: 120px;
  height: 40px;
  overflow: hidden;
  position: relative;
  margin-left: 42%;
  margin-bottom: -4%;
  margin-top: 2%;
}

.file-input-wrapper > input[type="file"] {
  position: absolute;
  top: 0;
  right: 0;
  opacity: 0;
  cursor: pointer;
}

.file-input-wrapper > .btn-file-input {
  display: inline-block;
  width: 120px;
  height: 40px;
  border: none;
  background-color: #1160bc;
  font-size: 85%;
  border-radius: 3px;
  color: #ffffff;
  font-weight: 400;
}

.uploadAttachmentHere {
  margin-left: 30%;
}

.uploadAttachmentText {
  color: #000000;
}
