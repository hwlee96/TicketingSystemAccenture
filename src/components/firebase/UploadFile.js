import React from "react";
// import axios from "axios"
import { storage } from "./firebaseExport";
import firebase from "firebase/app";

import LinearDeterminate from "../LinearDeterminate";

import "./UploadFile.css";

import cross from "./cross.svg";

class UploadFile extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      image: null,
      url: "",
      progress: 0
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleUpload = this.handleUpload.bind(this);
  }

  handleChange = e => {
    // console.log(e.target.files[0]);
    if (e.target.files[0]) {
      const image = e.target.files[0];
      this.setState(() => ({ image }));
    }
  };

  handleUpload = () => {
    const { image } = this.state;
    // console.log(`${this.props.userID}/${image.name}`);

    const uploadTask = storage
      .ref(`${this.props.userID}/${this.props.subject}/${image.name}`)
      .put(image);
    uploadTask.on(
      "state_changed",
      snapshot => {
        //progress fnc
        const progress = Math.round(
          (snapshot.bytesTransferred / snapshot.totalBytes) * 100
        );
        this.setState({ progress });
      },
      error => {
        //error fnc
        console.log("ERROR!!");
        console.log(error);
      },
      () => {
        //complete fnc
        storage
          .ref(`${this.props.userID}/${this.props.subject}`)
          .child(image.name)
          .getDownloadURL()
          .then(url => {
            // console.log(url);
            this.setState({ url });

            //lets add this url to database also

            firebase
              .database()
              .ref(`${this.props.userID}/${this.props.subject}/main`)
              .set(url);
          });
      }
    );
  };

  render() {
    // console.log(this.props.clickBack);

    let showDone = this.state.progress === 100;

    return (
      <div>
        <button
          type="button"
          className="buttonStyleCross"
          id="transparentButton"
          onClick={this.props.clickBack}
        >
          <img src={cross} alt="cross" />
        </button>

        <br />

        {this.state.url ? (
          <img
            src={this.state.url}
            alt="uploaded image"
            height="400"
            width="250"
            className="imageAttachment"
          />
        ) : null}
        <br />
        <div className="successfulFileAttach">
          {showDone ? <p>Image attached succesfully!</p> : null}
        </div>
        <br />
        {/* <progress id="uploader" value={this.state.progress} max="100" /> */}

        {/* <div class="btn-file-input">
          <input type="file" onChange={this.handleChange} />
        </div> */}
        <div className="uploadAttachmentHere">
          <p className="uploadAttachmentText">
            Upload an image attachment here:
          </p>
        </div>

        <div className="file-input-wrapper">
          <button className="btn-file-input">Choose Image</button>
          <input type="file" onChange={this.handleChange} />
        </div>

        <br />
        <br />
        <button className="buttonStyleUpload" onClick={this.handleUpload}>
          Attach
        </button>

        <LinearDeterminate id="uploader" progress={this.state.progress} />
      </div>
    );
  }
}
export default UploadFile;
