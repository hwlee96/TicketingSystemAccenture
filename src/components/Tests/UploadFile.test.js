import React from "react";
import renderer from "react-test-renderer";
import { shallow, mount, render, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import UploadFile from "../firebase/UploadFile";

configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const tree = renderer.create(<UploadFile />).toJSON();
  expect(tree).toMatchSnapshot();
});
