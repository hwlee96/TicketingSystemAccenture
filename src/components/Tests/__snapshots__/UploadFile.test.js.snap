// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    className="buttonStyleCross"
    id="transparentButton"
    type="button"
  >
    <img
      alt="cross"
      src="cross.svg"
    />
  </button>
  <br />
  <br />
  <div
    className="successfulFileAttach"
  />
  <br />
  <div
    className="uploadAttachmentHere"
  >
    <p
      className="uploadAttachmentText"
    >
      Upload an image attachment here:
    </p>
  </div>
  <div
    className="file-input-wrapper"
  >
    <button
      className="btn-file-input"
    >
      Choose Image
    </button>
    <input
      onChange={[Function]}
      type="file"
    />
  </div>
  <br />
  <br />
  <button
    className="buttonStyleUpload"
    onClick={[Function]}
  >
    Attach
  </button>
  <div
    className="LinearDeterminate-root-1"
  >
    <div
      aria-valuenow={0}
      className="MuiLinearProgress-root-2 MuiLinearProgress-colorPrimary-3 MuiLinearProgress-determinate-5"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar-12 MuiLinearProgress-barColorPrimary-13 MuiLinearProgress-bar1Determinate-16"
        style={
          Object {
            "transform": "scaleX(0)",
          }
        }
      />
    </div>
  </div>
</div>
`;
